plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobContainer.java

```diff
@@ -127,31 +129,67 @@ public void deleteBlob(String blobName) throws IOException {
     }
 
     @Override
-    public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) throws IOException {
-        if (blobNames.isEmpty()) {
-            return;
+    public void delete() throws IOException {
+        PlainActionFuture<Void> result = PlainActionFuture.newFuture();
+        asyncDelete(result);
+        try {
+            result.actionGet();
+        } catch (Exception e) {
+            throw new IOException("Exception during container delete", e);
         }
-        final PlainActionFuture<Collection<Void>> result = PlainActionFuture.newFuture();
-        final GroupedActionListener<Void> listener = new GroupedActionListener<>(result, blobNames.size());
-        final ExecutorService executor = threadPool.executor(AzureRepositoryPlugin.REPOSITORY_THREAD_POOL_NAME);
-        // Executing deletes in parallel since Azure SDK 8 is using blocking IO while Azure does not provide a bulk delete API endpoint.
-        // TODO: Upgrade to newer non-blocking Azure SDK 11 and execute delete requests in parallel that way.
-        for (String blobName : blobNames) {
-            executor.submit(new ActionRunnable<>(listener) {
-                @Override
-                protected void doRun() throws IOException {
-                    deleteBlobIgnoringIfNotExists(blobName);
-                    listener.onResponse(null);
-                }
-            });
+    }
+
+    private void asyncDelete(ActionListener<Void> listener) throws IOException {
```

Comment:
why recurse by level here and not list all sublevels at once?

Reply:
I'm assuming you're talking about flat listing all the blobs and then just deleting one by one like I did for GCS and S3?:
Mainly just to keep the changes size down a little for now as that didn't get us much here: The gain of being more effective when listing here is kinda small on Azure since we delete one by one (so the time that goes into listing requests is a lot less relevant overall than it would be if we deleted 100 or 1k blobs at a time). I can do a  bit of a bigger refactoring though and move the parallelization logic into the blob store. Let me look into that :)